babel.dependencies.config.js

```diff
@@ -1,31 +0,0 @@
-/** @format */
-const config = {
-	// see https://github.com/webpack/webpack/issues/4039#issuecomment-419284940
-	sourceType: 'unambiguous',
```

Comment:
This is the one difference I'm the least sure about. See the above link for documentation on what this does.

My hope is that every dependency on our current whitelist is behaving, thus allowing us to drop this: https://github.com/Automattic/wp-calypso/blob/948f23d6a0c7d9243377cb5346997238a160c183/webpack.config.js#L104-L111

Reply:
I'm not very clear on this one either. It sounds like it couldn't hurt to add it to the default configuration?

find`import`/`export` -> treat as esm
otherwise -> treat as cjs

That seems like desirable behavior.

Reply:
> otherwise -> treat as cjs

Is that what it does? Docs say "Parse the file using the ECMAScript Script grammar" -- that's not equivalent to CommonJS, is it?

For reference: https://babeljs.io/docs/en/options#sourcetype

> `unambiguous` can be quite useful in contexts where the type is unknown, but it can lead to false matches because it's perfectly valid to have a module file that does not use `import`/`export` statements.

:thinking:

Reply:
How common are modules that don't export anything? What use are they? They can only produce side effects because other modules can't access anything to use. Is there a subtlety that I'm missing?

Are modules without import/export common? Would it result in false matches for anything in Calypso or our other code bases?

Reply:
From the comment linked: 
> You're using useBultins: 'entry'/'usage, or @babel/plugin-transform-runtime, and you are running Babel on CommonJS files (either your own, or in random node_modules, if you're trying to compile that). Babel assumes files are ES modules by default, meaning those transforms would insert import statements into your file, triggering case 1. above. You can avoid this issue by setting sourceType: "unambiguous" in your Babel configuration, which will tell it to guess the type, like Webpack does, instead of assuming all files are modules.

We're running babel on commonJS files (`debug`: [package](https://github.com/visionmedia/debug/blob/master/package.json#L49), [entry module](https://github.com/visionmedia/debug/blob/master/src/index.js)), which, iirc, ends up inserting imports for transform-runtime, which breaks webpack.

To look at this another way, we're not transpiling to get from ESM to CommonJS. We're transpiling to get from ES6+ back to ES5. The packages on the list can be in either package format (or both!), but they're all shipping ES6+ features.

Reply:
That said, this works. :) So debug probably doesn't trigger transform-runtime includes currently.

I think this was added back when I was trying to transpile _everything_ in node modules, and some packages did trigger the inclusion of imports. If we keep the whitelist, then maybe we can continue to get away with that.

However the whitelist is application specific and probably doesn't belong in a generic build tool like calypso-build...

Reply:
> My hope is that every dependency on our current whitelist is behaving, thus allowing us to drop this:

What does "behaving" mean here? Are you talking about dropping the whitelist or dropping the sourceType?

Reply:
> > My hope is that every dependency on our current whitelist is behaving, thus allowing us to drop this:
> 
> What does "behaving" mean here? Are you talking about dropping the whitelist or dropping the sourceType?

Dropping the `sourceType` (sorry for being unclear) -- thus, behaving meaning that they transpile fine without `sourceType: unambiguous` :grimacing: